hazelcast-client-new/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -441,35 +464,37 @@ public void set(K key, V value, long ttl, TimeUnit timeunit) {
         final Data keyData = toData(key);
         final Data valueData = toData(value);
         invalidateNearCache(keyData);
-        MapSetRequest request = new MapSetRequest(name, keyData, valueData,
+        ClientMessage clientMessage = MapSetParameters.encode(name, keyData, valueData,
                 ThreadUtil.getThreadId(), getTimeInMillis(ttl, timeunit));
-        invoke(request, keyData);
+
+        invoke(clientMessage, keyData);
     }
 
     @Override
     public void lock(K key) {
         checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
         final Data keyData = toData(key);
-        MapLockRequest request = new MapLockRequest(name, keyData, ThreadUtil.getThreadId());
-        invoke(request, keyData);
+        ClientMessage clientMessage = MapLockParameters.encode(name, keyData, ThreadUtil.getThreadId());
+        invoke(clientMessage, keyData);
     }
 
     @Override
     public void lock(K key, long leaseTime, TimeUnit timeUnit) {
         checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
         final Data keyData = toData(key);
-        MapLockRequest request = new MapLockRequest(name, keyData,
-                ThreadUtil.getThreadId(), getTimeInMillis(leaseTime, timeUnit), -1);
-        invoke(request, keyData);
+        ClientMessage clientMessage = MapLockWithLeaseTimeParameters.encode(name, keyData,
+                ThreadUtil.getThreadId(), getTimeInMillis(leaseTime, timeUnit));
+        invoke(clientMessage, keyData);
     }
 
     @Override
     public boolean isLocked(K key) {
         checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
         final Data keyData = toData(key);
-        MapIsLockedRequest request = new MapIsLockedRequest(name, keyData);
-        Boolean result = invoke(request, keyData);
-        return result;
+        ClientMessage clientMessage = MapIsLockedParameters.encode(name, keyData, ThreadUtil.getThreadId());
```

Comment:
Isn't it easier to call it request and response instead of 'clientMessage' and 'resultMessage'. This provides more information which message is used for the request and which message contains the response.